packages/upload/src/vaadin-upload-mixin.js

```diff
@@ -159,6 +159,7 @@ export const UploadMixin = (superClass) =>
           type: Array,
           notify: true,
           value: () => [],
+          sync: true,
```

Comment:
Added `sync` to several properties mainly to preserve existing behavior when adding files and reduce changes related to tests. Other properties do not necessarily need this change.